Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift

```diff
@@ -106,30 +106,32 @@ extension BinarySearchTree {
   */
   @discardableResult public func remove() -> BinarySearchTree? {
     let replacement: BinarySearchTree?
-    
+
     // Replacement for current node can be either biggest one on the left or
     // smallest one on the right, whichever is not nil
-    if let left = left {
-      replacement = left.maximum()
-    } else if let right = right {
+    if let right = right {
```

Comment:
The previous [version](https://github.com/raywenderlich/swift-algorithm-club/blob/ca35d57d14073ac99f6a74c5a2057f1ab7f28c60/Binary%20Search%20Tree/Solution%201/BinarySearchTree.playground/Sources/BinarySearchTree.swift#L135-L139) before 942e3b1 use successor to replace the node with 2 children. The unit tests are also based on this logic. I am kind of lazy to modify all the test cases 😵.